boa/src/builtins/string/mod.rs

```diff
@@ -140,9 +141,11 @@ impl String {
             None => RcString::default(),
         };
 
-        let length = string.encode_utf16().count();
-
-        this.set_field("length", Value::from(length as i32));
+        let length = DataDescriptor::new(
+            Value::from(string.encode_utf16().count() as i32),
```

Comment:
I know it was already there, but I think we should remove the `as i32`, `From<usize>` is implemeted for `Value` so the conversion is unnecessary.
```suggestion
            Value::from(string.encode_utf16().count()),
```